packages/indexer/src/configs/schema.graphql

```diff
@@ -980,6 +980,182 @@ type OrderStatusMetadata @entity {
 	createdAt: Date! @index
 }
 
+"""
+Represents an order placed through the IntentGatewayV2 contract.
+This entity tracks cross-chain orders for V2 of the IntentGateway.
+"""
+type OrderV2 @entity {
+	"""
+	Unique identifier for the order (commitment)
+	"""
+	id: ID!
+
+	"""
+	Address of the user who placed the order
+	"""
+	user: String!
+
+	"""
+	Source chain identifier where the order originates from
+	"""
+	sourceChain: String!
+
+	"""
+	Destination chain identifier where the order will be executed
+	"""
+	destChain: String!
+
+	"""
+	Unique commitment hash of the order used for verification
+	"""
+	commitment: String!
+
+	"""
+	Timestamp after which the order expires
+	"""
+	deadline: BigInt!
+
+	"""
+	Sequential number to ensure uniqueness of the order
+	"""
+	nonce: BigInt!
+
+	"""
+	Total fees supplied by the user for executing the order
+	"""
+	fees: BigInt!
+
+	"""
+	Optional session key used to select winning solver
+	"""
+	session: String
+
+	"""
+	Array of predispatch token addresses
+	"""
+	predispatchTokens: [String!]!
+
+	"""
+	Array of predispatch token amounts corresponding to predispatchTokens
+	"""
+	predispatchAmounts: [BigInt!]!
```

Comment:
think we should make this a struct instead

Reply:
A struct in a schema?

Reply:
Yeah kinda like in `statusMetadata`

Reply:
Linked schema i think.